### PR TITLE
Analysis options integration.

### PR DESCRIPTION
**Pull request by pq**
This integrates the linter with the analyzer's options mechanism.

### CODE REVIEW

**Comment by pq** [comment]
Note: there's a bit of refactoring in the works here to cleanup the way that rule defaults are set.  (I'm leaning towards an actual default options file to specify what rules are enabled by default.)  IOW: maybe best to ignore weirdness around defaults.  I'll fix that in a follow-up.

**Comment by pq** [comment]
@bwilkerson PTAL.

File: lib/src/config.dart [hunk 1]
```diff
@@ -4,11 +4,51 @@
...
+  var yaml = loadYamlNode(fileContents, sourceUrl: fileUrl);
+  if (yaml is YamlMap) {
+    return _parseConfig(yaml);
+  }
+  return null;
+}
+
+LintConfig _parseConfig(Map optionsMap) {
+  if (optionsMap != null) {
+    var options = optionsMap['linter'];
```
Comment (bwilkerson) [inline lib/src/config.dart]: We might want to encourage (and use) a convention of fully qualified top-level names, such as 'org.dartlang.linter'.

File: lib/src/linter.dart [hunk 2]
```diff
@@ -154,7 +154,9 @@ class LinterOptions extends DriverOptions {
     }
   }
   void configure(LintConfig config) {
-    enabledLints = ruleRegistry.enabled(config);
+    // TODO(pquitslund): revisit these default-to-on semantics.
+    enabledLints = ruleRegistry.where((LintRule rule) =>
+        !config.ruleConfigs.any((rc) => rc.disables(rule.name)));
```
Comment (bwilkerson) [inline lib/src/linter.dart]: This condition was reversed in src/rules.dart. Should it have been changed here as well?

**Comment by bwilkerson** [comment]
LGTM

File: lib/src/linter.dart (hunk 2 above)
Comment (pq) [inline lib/src/linter.dart]: Nope.  That was deliberate.  The current semantics are actually I think **wrong**.  I'll clean that up in a follow-up CL.

File: lib/src/config.dart (hunk 1 above)
Comment (pq) [inline lib/src/config.dart]: Assuming options processors are associated with plugins and that plugins will be published via pub I was thinking that the pub namespace would solve this for us.  Does that sound right?

Comment (bwilkerson) [inline lib/src/config.dart]: I'm not sure that we should assume that all plugins will be written as packages, or that if they are they will be published on pub. But it does seem reasonable to say that plugins that meet those criteria should use the package name and that others should use a fully qualified name.

Comment (bwilkerson) [inline lib/src/config.dart]: Whatever it is, we should document the convention somewhere.

**Pull request merged**